aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -50,7 +52,24 @@
    * @param location fully qualified URI
    */
   S3URI(String location) {
+    this(location, ImmutableMap.of());
+  }
+
+  /**
+   * Creates a new S3URI in the form of scheme://(bucket|accessPoint)/key?query#fragment with additional information
+   * on accessPoints.
+   * <p>
+   * The URI supports any valid URI schemes to be backwards compatible with s3a and s3n,
+   * and also allows users to use S3FileIO with other S3-compatible object storage services like GCS.
+   * If the accessPoints contains a mapping of the given bucket used in location then the corresponding accessPoint
```

Comment:
nit: this sentence is a bit convoluted, what about "if the bucket of the location has an access point in the mapping, the access point is used to perform all the S3 operations"